pkg/mcp/loader.go

```diff
@@ -508,6 +509,89 @@ func (sm *SessionManager) updatedMCPPodName(ctx context.Context, url, id string,
 	}
 }
 
+// LaunchTemporaryInstance creates a temporary MCP server from a catalog entry, lists its tools,
+// then shuts it down and returns the tool preview data.
+func (sm *SessionManager) LaunchTemporaryInstance(ctx context.Context, catalogEntryManifest otypes.MCPServerCatalogEntryManifest, userURL string, configEnv map[string]string) ([]otypes.MCPServerTool, error) {
+	// Convert catalog entry to server manifest
+	serverManifest, err := otypes.MapCatalogEntryToServer(catalogEntryManifest, userURL)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert catalog entry to server config: %w", err)
+	}
+
+	// Create temporary MCPServer object to use existing conversion logic
+	tempName := "temp-preview-" + hash.Digest(serverManifest)[:8]
+	tempMCPServer := &v1.MCPServer{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: tempName,
+		},
+		Spec: v1.MCPServerSpec{
+			Manifest: serverManifest,
+		},
+	}
+
+	// Use existing ServerToServerConfig function
+	serverConfig, missingFields, err := ServerToServerConfig(*tempMCPServer, "temp", configEnv)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create server config: %w", err)
+	}
+
+	if len(missingFields) > 0 {
+		return nil, fmt.Errorf("missing required configuration fields: %v", missingFields)
+	}
+
+	// Create MCP client and list tools
+	client, err := sm.ClientForServer(ctx, "system", serverManifest.Name, tempName, serverConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create MCP client: %w", err)
+	}
+
+	// Ensure cleanup happens regardless of success or failure
+	defer func() {
+		if cleanupErr := sm.ShutdownServer(ctx, serverConfig); cleanupErr != nil {
+			fmt.Printf("failed to cleanup temporary instance %s: %v\n", tempName, cleanupErr) // TODO: use log package
```

Comment:
reminder to switch over to the log package

Reply:
Thanks, forgot to do this

Reply:
Fixed